libraries/botbuilder-dialogs/src/dialogContext.ts

```diff
@@ -191,14 +191,20 @@ export class DialogContext {
      * Obtain the CultureInfo in DialogContext.
      * @returns a locale string.
      */
-    public getLocale(): string
-    {
-        let locale: string = undefined;
-        if (this.context.locale !== undefined && this.context.locale !== null) {
-            locale = this.context.locale;
+    public getLocale(): string {
+        const _turnLocaleProperty = 'turn.locale';
+
+        const _trunLocaleValue = this.state.getValue(_turnLocaleProperty);
+        if (_trunLocaleValue !== undefined && _trunLocaleValue !== '') {
+            return _trunLocaleValue;
+        }
+
+        const locale = this.context.activity?.locale;
+        if (locale !== undefined) {
```

Comment:
in .NET, here is 
```cs
if (!string.IsNullOrEmpty(locale)). 
```
if we keep the same here, it will break some tests in LGGenerator.test.js

Reply:
How is the environment locale set in .NET? Since we don't have threads or any other natural isolation, setting values on a context parameter (i.e. `this.context`) is likely the cleanest way to achieve the same functionality.

Reply:
In .NET, it is
```cs
Thread.CurrentThread.CurrentCulture = new CultureInfo("es-ar");
```

Or maybe we can slightly modify the tests in javascript since it is hard to modify the environment locale.

Reply:
I guess I'm wondering, semantically, if this is something we can really do in Javascript. There is no notion of threads or domains (well, they're deprecated anyways) to support collecting context or culture. We have functions and arguments, and we already have a context argument that gets passed around. Is that context parameter not sufficient?

Reply:
I think the current workaround is OK. An empty string in locale, both LG and expression will fallback to the correct locale resolution.